advanced_source/torch-script-parallelism.rst

```diff
@@ -1,281 +1,3 @@
-
-This snippet of code will write out a file named ``parallel.json``. If you
-navigate Google Chrome to ``chrome://tracing``, click the ``Load`` button, and
-load in that JSON file, you should see a timeline like the following:
-
-.. image:: https://i.imgur.com/rm5hdG9.png
-
-The horizontal axis of the timeline represents time and the vertical axis
-represents threads of execution. As we can see, we are running two ``lstm``
-instances at a time. This is the result of our hard work parallelizing the
-bidirectional layers!
-
-Parallelizing Models in the Ensemble
-------------------------------------
-
-You may have noticed that there is a further parallelization opportunity in our
-code: we can also run the models contained in ``LSTMEnsemble`` in parallel with
-each other. The way to do that is simple enough, this is how we should change
-the ``forward`` method of ``LSTMEnsemble``:
-
-.. code-block:: python
-
-        def forward(self, x : torch.Tensor) -> torch.Tensor:
-            # Launch tasks for each model
-            futures : List[torch.jit.Future[torch.Tensor]] = []
-            for model in self.models:
-                futures.append(torch.jit.fork(model, x))
-
-            # Collect the results from the launched tasks
-            results : List[torch.Tensor] = []
-            for future in futures:
-                results.append(torch.jit.wait(future))
-
-            return torch.stack(results).sum(dim=0)
-
-Or, if you value brevity, we can use list comprehensions:
-
-.. code-block:: python
-
-        def forward(self, x : torch.Tensor) -> torch.Tensor:
-            futures = [torch.jit.fork(model, x) for model in self.models]
-            results = [torch.jit.wait(fut) for fut in futures]
-            return torch.stack(results).sum(dim=0)
-
-Like described in the intro, we've used loops to fork off tasks for each of the
-models in our ensemble. We've then used another loop to wait for all of the
-tasks to be completed. This provides even more overlap of computation.
-
-With this small update, the script runs in ``1.4`` seconds, for a total speedup
-of ``32%``! Pretty good for two lines of code.
-
-We can also use the Chrome tracer again to see where's going on:
-
-.. image:: https://i.imgur.com/kA0gyQm.png
-
-We can now see that all ``LSTM`` instances are being run fully in parallel.
-
-Conclusion
-----------
-
-In this tutorial, we learned about ``fork()`` and ``wait()``, the basic APIs
-for doing dynamic, inter-op parallelism in TorchScript. We saw a few typical
-usage patterns for using these functions to parallelize the execution of
-functions, methods, or ``Modules`` in TorchScript code. Finally, we worked through
-an example of optimizing a model using this technique and explored the performance
-measurement and visualization tooling available in PyTorch.
+.. warning::
+    TorchScript is deprecated, please use 
+    `torch.export<https://docs.pytorch.org/tutorials/intermediate/torch_export_tutorial.html>`_ instead.
```

Comment:
broken in preview: https://docs-preview.pytorch.org/pytorch/tutorials/3453/advanced/torch-script-parallelism.html
```suggestion
    `torch.export <https://docs.pytorch.org/tutorials/intermediate/torch_export_tutorial.html>`_ instead.
```